app/Domain/Integrations/Controllers/IntegrationController.php

```diff
@@ -80,6 +81,8 @@ public function store(StoreIntegration $storeIntegration): RedirectResponse
 
         $this->integrationRepository->save($integration);
 
+        IntegrationCreated::dispatch($integrationId);
```

Comment:
Maybe it's a bit better if the `IntegrationRepository` would dispatch these messages? Since the `save()` method on it could in theory be called from multiple places and not just this controller

Reply:
Yes! That is indeed a very good approach. Most Laravel examples use the Eloquent model directly so they suffer from this copy/paste issue. We have a central place to do the saving so we can do the dispatch there.

Note: the Laravel Nova code still needs the dispatch because it bypasses the repo.

Reply:
As mentioned on Slack we could use the Eloquent events so it also works in Nova https://laravel.com/docs/9.x/eloquent#events

Reply:
I refactored this to use the Eloquent events and it works smoothly. This is indeed the best approach to handle our domain events.